lib/sles4sap_publiccloud.pm

```diff
@@ -434,16 +437,46 @@ sub wait_for_pacemaker {
      "yes" - sets default SBD value which is calculated from a formula
      "<number of seconds>" - sets sepcific delay in seconds
 
+     Returns integer representing wait time.
+
 =cut
 
 sub setup_sbd_delay() {
-    my ($self, $set_delay) = @_;
-    my $delay = $set_delay || "no";
-    record_info("SBD delay", "Setting SBD delay to: $delay");
+    my ($self) = @_;
+    my $delay = get_var('HA_SBD_START_DELAY');
+    $delay =~ s/[^ye]s//g;
+    if (get_var('HA_SBD_START_DELAY') eq '') { record_info('SBD delay', 'Skipping, parameter without value'); return; }
+
+    croak("<\$set_delay> value must be either 'yes', no or an integer. Got value: $delay")
+      unless looks_like_number($delay) or grep /^$delay$/, qw(yes no);
+
     $self->cloud_file_content_replace('/etc/sysconfig/sbd', '^SBD_DELAY_START=.*', "SBD_DELAY_START=$delay");
+    record_info('SBD delay', "SBD delay set to: $delay");
     return 1;
 }
 
+=head2 sbd_delay_formula
+    sbd_delay_formula();
+
+
+=cut
+
+sub sbd_delay_formula() {
+    my ($self) = @_;
+    my %params = (
+        'corosync_token' => $self->run_cmd(cmd => "corosync-cmapctl | awk -F \" = \" '/runtime.config.totem.token\\s/ {print \$2 /1000}'"),
```

Comment:
is this code portion similar to https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/17145/files#diff-f1dc32f4e8e57e2753021dd0a58d9cbd5cc8a10f9f919280ca8e75ab6d9ed9b1R986 ?

Reply:
Yep. The difference is that PC uses "run_cmd" and on prem tests are doing "script_output". 
I guess I could bind that somehow together, but didn't wanna complicate things and let the function accept the values.